gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/v4/subscription/context/SubscriptionRequest.java

```diff
@@ -146,31 +160,37 @@ public long timestamp() {
 
     @Override
     public String remoteAddress() {
-        throw new IllegalStateException();
+        // Subscription request is issued by the gateway itslef.
+        return "localhost";
     }
 
     @Override
     public String localAddress() {
-        throw new IllegalStateException();
+        // Subscription request is issued by the gateway internally.
+        return "localhost";
     }
 
     @Override
     public SSLSession sslSession() {
-        throw new IllegalStateException();
+        // There is no ssl session on a subscription request.
+        return null;
     }
 
     @Override
     public Metrics metrics() {
-        return null;
+        // We still maintain metrics to be compliant with the execution flow (eg: policies), that could access metrics.
+        return metrics;
     }
 
     @Override
     public boolean ended() {
-        return false;
+        // Subscription request must always be considered has ended because it is issued by the gateway itself.
+        return true;
```

Comment:
Don't get the why. Could you elaborate?

Reply:
request.end() indicates the request has been fully consumer (including the request body). Because the subscription request is not a regular http request, there is no body to consume so we can consider the request is already ended (not the case for the response). Sounds good to you ?